doc/release-notes-28048.md

```diff
@@ -0,0 +1 @@
+- The `-stopatheight` option will now stop exactly at the specified height, rather than at or above the specified height. Since the node shuts down as soon as it find a valid chain at the specified height, it is possible for the resulting chain to no longer be the most-work chain, if some of the blocks above the specified height are invalid.
```

Comment:
Just wanted to note that this proposal has some history, see Issue #13477 and the two closed PRs: #13490 and #13713.

In particular, comment https://github.com/bitcoin/bitcoin/pull/13490#issuecomment-398069237 still seems relevant:
The expectation that -stopatheight stops exactly at the specified height relies on `ActivateBestChainStep()` only connecting a single block in each invocation, which is currently the case, but more 
of a coincidence due to not wanting to lock `cs_main` for too long than an actual guarantee given by the function.

Maybe it would be good to explicitly document in `ActivateBestChainStep` that `-stopatheight` relies on it conecting just a single block (outside of a reorg situation)?

Reply:
I looked into this a bit deeply and did some test runs on signet, and now I'm no longer sure that this is a behavioral change requiring a release note in the first place:

My understanding is that the IBD behavior on master is:
* `ActivateBestChainStep()`  will connect only one block if we made progress
* The inner do / while loop of `ActivateBestChain` also will only run once unless the chain has less work than intitially (only relevant in a reorg).
* so during IBD we'll connect one block, release `cs_main`, call `StartShutdown()` if the stopheight is reached, break out of the outer do / while loop because `m_chainman.m_interrupt` is set.
* There is a race between `Shutdown()` executed in the `init` thread, and `msghand`, so it's possible that `msghand` connects another block or two (by executing new calls to ABC!) before it's being stopped by `Shutdown()`. But the same should be possible after this PR?

Also, the explanation of previous behavior in the commit msg ("It would not stop connecting blocks after the specified  height if they were already downloaded.") does not seem to be accurate because of the existing [break](https://github.com/bitcoin/bitcoin/blob/357e3f6aa476658aecae7239b4b06d2bc362ff1e/src/validation.cpp#L3191C13-L3191C23).

Reply:
> Naively, I'd think ProcessNewBlock should just do if (m_interrupt) return false; and the types of races you mention would be avoided.

Hmm, if it arrived to `ProcessNewBlock` is because the node requested the block. So I would try to not discard it, and instead make it pass through `AcceptBlock` so it can be stored. Then quit early before ABC.